app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java

```diff
@@ -50,13 +63,24 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
                 WindowManager.LayoutParams.FLAG_FULLSCREEN);
         setTitle(R.string.mem_graph);
 
-        MainActivity mainActivity = new MainActivity();
-        rawData = mainActivity.getDeviceData();
-        if(rawData != null)
-        parseDataForGraph(rawData);
-
         memGraph = findViewById(R.id.mem_graph);
 
+        LayoutInflater layoutInflater = getLayoutInflater();
+        View progressView = layoutInflater.inflate(R.layout.progress_dialog_layout, null);
```

Comment:
Can we use the stock progress dialog without creating our own?

Reply:
@CloudyPadmal The `ProgressDialog` is deprecated. I guess we need to make a layout. In this case, I have made an simple one with just the progress bar and a text. Please let me know otherwise.

Reply:
@adityastic will you be able to design us a nice progress dialog if you get time? :)

Reply:
@CloudyPadmal Have a look at this https://github.com/d-max/spots-dialog I used it earlier it has nice a animation too. :D

Reply:
It doesn't look like Android and we don't need a lot of dependencies especially for a progress dialog.